xlsx_reader.py

```diff
@@ -0,0 +1,185 @@
+    else:
+        authors_names = get_full_name(authors)
+
+    return authors_names
+
+
+# reads book's data from file
+def get_book_data(file_location):
+    book_list = []
+    workbook = load_file(file_location)
+
+    # excluding sheets with unnecessary data
+    for sheet_index in range(workbook.nsheets - 2):
+        current_sheet = workbook.sheet_by_index(sheet_index)
+        rows = current_sheet.nrows
+
+        # excluding data from the title of the column
+        for row_index in range(1, rows):
+            current_shelf = current_sheet.name
+            title = (current_sheet.cell_value(row_index, 1)).strip()
+            authors = current_sheet.cell_value(row_index, 2)
+            author = get_authors_data(authors)
+
+            if current_shelf == 'General':
+                user = current_sheet.cell_value(row_index, 4)
+                date_of_rental = current_sheet.cell(row_index, 5)
+                status = current_sheet.cell(row_index, 5)
+                author = get_authors_data(authors)
+                asset = current_sheet.cell_value(row_index, 3)
+                book_properties = {'authors': author,
+                                   'title': title,
+                                   'asset': asset,
+                                   'user': user}
+                book_list.append(book_properties)
+
+            else:
+                asset = current_sheet.cell_value(row_index, 3)
+                book_properties = {'authors': author,
+                                   'current_shelf': current_shelf,
+                                   'title': title,
+                                   'asset': asset,
+                                   'user': user,
+                                   'date_of_rental': date_of_rental,
+                                   'status': status}
+                book_list.append(book_properties)
+
+    return book_list
+
+
+# reading magazine's data from file
+def get_magazine_data(file_location):
+    workbook = load_file(file_location)
+    magazines_list = []
+    current_sheet = workbook.sheet_by_index(2)
+    rows = current_sheet.nrows
+
+    # reading rows except the title of the column
+    for row_index in range(1, rows):
+        title = (current_sheet.cell_value(row_index, 1)).strip()
+        year = current_sheet.cell_value(row_index, 2)
+        issue = current_sheet.cell_value(row_index, 3)
+        magazine_properties = {'title': title, 'year': year, 'issue': issue}
+        magazines_list.append(magazine_properties)
+    return magazines_list
+
+
+# writing authors, books and copies data in database
+def get_books(file_location):
+    books_properties = get_book_data(file_location)
+    asset_codes = []
+
+    for book in books_properties:
+        title = book['title']
+        asset = book['asset']
+        asset_codes.append(asset)
+        authors = book['authors']
+        list_of_authors = []
+
+        if type(authors) is tuple:
```

Comment:
there is a function for stuff like this:
```isinstance()```
https://docs.python.org/3/library/functions.html#isinstance
i.e. ```if isinstance(authors, tuple):
            do_stuff()```

Reply:
thanks a lot for an advice!